### PR TITLE
(LD-372) develop the RestaurantHeader component

### PR DESCRIPTION
**Pull request by snoattoh**

### CODE REVIEW

**Comment by snoattoh** [comment]
This taking a bit longer than expected, converting the css to be used with React Native is part of a learning experience.

File: assets/Typography.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  // | "body1"
+  // | "body2"
+  // | "button"
+  // | "caption"
+  // | "overline";
+};
+
+export const Typography: React.FC<Props> = ({ variant, children }) => {
+  return <Text style={styles[variant]}>{children}</Text>;
+};
```
Comment (snoattoh) [inline assets/Typography.tsx]: Where should helpers like this live in the repo? Probably not in assets?

File: src/components/RestaurantHeader/RestaurantHeader.test.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    expect(
+      screen.getByText(restaurantWithAddress.address.street, { exact: false })
+    ).toBeOnTheScreen();
+    expect(
+      screen.getByText(
+        `${restaurantWithAddress.address.city}, ${restaurantWithAddress.address.state} ${restaurantWithAddress.address.zip}`,
+        { exact: false }
+      )
+    ).toBeOnTheScreen();
+  });
```
Comment (snoattoh) [inline src/components/RestaurantHeader/RestaurantHeader.test.tsx:76]: Not terribly familiar with the differences between testing React Native and using toBeOnTheScreen, but if anyone has a better/stronger suggestion or solution I'm open to any improvement here.

File: src/components/RestaurantHeader/RestaurantHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    marginBottom: 15,
+    color: "#fff",
+    textShadowColor: "rgba(0, 0, 0, 0.5)",
+    textShadowOffset: { width: 0, height: -1 },
+    textShadowRadius: 0,
+  },
+  background: {},
+  address: {},
+  hoursPrice: {},
+  openNow: {},
```
Comment (snoattoh) [inline src/components/RestaurantHeader/RestaurantHeader.tsx:52]: Empty for now, wondering if there's merit to opening styling as a seperate ticket or if there's a preference I put at least passable styling now.

**Review by snoattoh — COMMENTED** [review]
Just a few requests for feedback.

File: assets/Typography.tsx (hunk 1 above)
Comment (rjspencer) [reply assets/Typography.tsx]: Go with a `components` folder and a fractal modlet pattern

`/components/Typography/Typography.tsx`
`/components/Typography/Typography.test.tsx` 
`/components/Typography/index.ts`

File: src/components/RestaurantHeader/RestaurantHeader.test.tsx (hunk 2 above)
Comment (rjspencer) [reply src/components/RestaurantHeader/RestaurantHeader.test.tsx:76]: Normally I would do a partial match with a regex, but I didn't know about `{ exact: false }`. Makes sense here when using variables.

File: src/components/RestaurantHeader/RestaurantHeader.tsx (hunk 3 above)
Comment (rjspencer) [reply src/components/RestaurantHeader/RestaurantHeader.tsx:52]: Don't worry about the styling too much. Once we have a few more things in the app we can make something sensible for mobile.

File: src/components/Typography/Typography.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    marginTop: 2.33,
+    marginBottom: 2.33,
+    marginLeft: 0,
+    marginRight: 0,
+    fontWeight: "bold",
+  },
+});
+
+type Props = {
+  variant: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
```
Comment (rjspencer) [inline src/components/Typography/Typography.tsx:61]: I've been thinking about this and I don't like that this forces a web idea (h1, h2, etc) on RN, but on the other hand "heading" is a valid concept and having headings of different levels makes sense so why not "h1"? 

Point being... if you come up with better names for these variants at some point we should talk about it.

**Review by rjspencer — APPROVED** [review]
Looks good 🚀

File: src/components/Typography/Typography.tsx (hunk 4 above)
Comment (snoattoh) [reply src/components/Typography/Typography.tsx:61]: I'm gonna review the article a bit and see if there's an approach we can go with in terms of naming. 

I'll consult my gut as well. May get the idea while working on the other components.

**Pull request merged**